feat: add final task Stage 1 (#1462)
rolling-scopes-school/tasks: tasks/nonograms/README.md

**In the current task you have to implement only binary version of the game**

### Main functional requirements

`Basic (required):`

- initially, `body` in the index.html file must be empty (only script tag is allowed), **all necessary elements are generated using JS**
- the design should be adaptive (or responsive) from _(500px <= width)_. It is acceptable to change the appearance for the mobile version (for example, hide the buttons in the burger menu)
- the default size of the frame is 5x5. The clues are given at the top and left side of the grid. The sequence of numbers must be logically arranged and help the player solve the nonogram
- every 5 cells in a rows and columns (both player's field and clues) should be divided by divider (bold line).
- a player is able to fill in a square in the grid, using left mouse-click. On a click event, trigger the functionality of changing the color of the grid to dark (black). When player clicks on dark cell - it will change to empty (white).
- end game when players fill all **black** squares correctly according to the hints. On a successful game solution, display "Great! You have solved the nonogram in ## seconds!"

`Advanced:`

> - the game should have at least 5 templates for easy level (5x5) and the player is able to choose what picture he/she wants to solve (for example list of items)

Reviewer: ```suggestion
- the game should have at least 5 templates for easy level (5x5). Players should be able to choose the picture they wish to solve, possibly through a list of items.
```

> - a player is able to fill in a square in the grid changing the color of the grid to crossed-cell(X) using right mouse-click. Context menu should not appear. Filling empty cells with X is not obligatory to win the game.

Reviewer: ```suggestion
- players should have the ability to mark a square on the grid as a crossed-cell (X) using a right mouse-click. The context menu should be disabled for this action. Marking empty cells with an X is not mandatory to win the game.
```

> - the game should have at least 5 templates for easy level (5x5) and the player is able to choose what picture he/she wants to solve (for example list of items)
> - a player is able to fill in a square in the grid changing the color of the grid to crossed-cell(X) using right mouse-click. Context menu should not appear. Filling empty cells with X is not obligatory to win the game.
> - the game can be restarted (reset) without reloading the page (for example, by clicking on button `Reset game`). Otherwise the player can change game template or game level (for example, you could implement menu with options) without reloading the page.
> - display the game duration in format XX:XX, stop-watch will start after first click on field (not on clues)
> - the game should include sound effects for events such as mark a cell as black, flagging a cell as X, and game over (both win and lose).
> - the staging of the game is saved (for example, using localStorage), so that when player clicks on button "continue last game", he can continue playing from where he left off

Reviewer: ```suggestion
- The game should offer at least 5 templates for the easy level (5x5). Players should be able to choose the picture they wish to solve, possibly through a list of items.
- Players should have the ability to mark a square on the grid as a crossed-cell (X) using a right mouse-click. The context menu should be disabled for this action. Marking empty cells with an X is not mandatory to win the game.
- Include a feature to restart (reset) the game without needing to reload the page. This can be achieved, for instance, by clicking a `Reset game` button. Additionally, players should be able to change the game template or level (via a menu with options) without reloading the page.
- Display the game duration in the format XX:XX. The stopwatch should start after the player's first click on the field (not on the clues).
- Implement sound effects for specific events such as marking a cell as black, flagging a cell as X, and at the end of the game (both for winning and losing scenarios).
- Ensure the game's progress is saved, for example, using localStorage. This allows players to continue from where they left off when they click on a "continue last game" button.
```

`Additional (to get extra points):`

- dark/light themes of the game. Changing the theme implies a change in the entire color scheme of the application, including the background color, cell colors, counters, buttons etc.

> - the game should have 3 levels of difficulty (5x5, 10x10, 15x15) which changes the size of the game board and the main template (pictures) and at least 5 templates for each level. User should have ability to choose the picture that he wants to solve

Reviewer: ```suggestion
- Implement three levels of difficulty in the game: easy (5x5), medium (10x10), and hard (15x15). Each level should vary in the size of the game board and the complexity of the main template (pictures). 
- Provide at least five unique templates for each difficulty level. 
- Users should have the ability to select the specific picture they wish to solve from the available templates in each level.
```

- [ ] the game should end when the player reveals all cells correctly and related message is displayed at the end of the game: `+20`

### Advanced scope +90 points

- [ ] the game should have at least 5 templates for easy level (5x5) and the player is able to choose what picture he/she wants to solve. `+15`
- [ ] a player is able to fill in a square in the grid changing the color of the grid to crossed-cell(X) using right mouse-click. Context menu doesn't appear: `+20`
- [ ] the game can be restarted without reloading the page: `+15`
- [ ] game duration is displayed, stop-watch will start after first click on field (not on clues): `+10`
- [ ] sound accompaniment (on/off) for every events (see **Advanced** block): `+15`
- [ ] implemented saving the state of the latest game: `+15`

### Hacker scope +80 points

- [ ] option to choose different themes for the game board (dark/light themes): `+15`
- [ ] ability to change the size (5x5, 10x10, 15x15) is implemented and there are least 5 templates for each level: `+20`
- [ ] implemented saving the latest 5 win results with sorting: `+15`

> - [ ] "random game" button is implemented: `+15`

Reviewer: I think here we need to write not only a button but also the logic of a random game should be implemented.
